jme3-examples/src/main/java/jme3test/TestSceneBuilder.java

```diff
@@ -0,0 +1,1014 @@
+        Spatial tile;
+        if (v != null) {
+            tile = terrainSrc.getChild(name + FastMath.rand.nextInt(v)).clone();
+        } else {
+            tile = terrainSrc.getChild(name).clone();
+        }
+        Texture color = assetManager.loadTexture(new TextureKey("Scenes/TestScene/grid-grey.png", false));
+        color.setWrap(Texture.WrapMode.Repeat);
+        color.setMinFilter(Texture.MinFilter.BilinearNearestMipMap);
+        color.setMagFilter(Texture.MagFilter.Bilinear);
+        color.setAnisotropicFilter(Math.min(ANISOTROPIC_LEVEL, app.getRenderer().getLimits().get(Limits.TextureAnisotropy)));
+        Material mat = new Material(assetManager, "Common/MatDefs/Light/PBRLighting.j3md");
+        mat.setTexture("BaseColorMap", color);
+        mat.setFloat("Metallic", .1f);
+        mat.setFloat("Roughness", .9f);
+        tile.setMaterial(mat);
+        return tile;
+    }
+    private DirectionalLight getSunLight() {
+        for (Light l : node.getLocalLightList()) {
+            if (l.getName().equals("SunLight") && l instanceof DirectionalLight) {
+                return (DirectionalLight)l;
+            }
+        }
+        return null;
+    }
+    private FilterPostProcessor getOrCreateFpp() {
+        ViewPort vp = app.getViewPort();
+        for (SceneProcessor p : vp.getProcessors()) {
+            if (p instanceof FilterPostProcessor) {
+                return (FilterPostProcessor)p;
+            }
+        }
+        FilterPostProcessor fpp = new FilterPostProcessor(assetManager);
+        vp.addProcessor(fpp);
+        return fpp;
+    }
+    private BulletAppState getOrCreateBullet() {
+        BulletAppState bullet = app.getStateManager().getState(BulletAppState.class, false);
+        if (bullet == null) {
+            bullet = new BulletAppState();
+            app.getStateManager().attach(bullet);
+        }
+        return bullet;
+    }
+    private PhysicsSpace getOrCreatePhysics() {
+        return getOrCreateBullet().getPhysicsSpace();
+    }
+
+    /**
+     * Dedicated test application for {@link TestSceneBuilder}.
+     */
+    private static class TestApp extends SimpleApplication {
+
+        private static void start(String[] args) {
+            new TestApp().start();
+        }
+
+        @Override
+        public void simpleInitApp() {
+
+            TestSceneBuilder scene = new TestSceneBuilder(this);
+            scene.configure();
+            scene.baseScene();
+            scene.brightMountainsSun();
+            scene.hardwareProbe();
+            scene.brightMountainsSky();
+            scene.softBloom(b -> b.setGlowFactor(0.1f));
+            scene.scattering();
+            scene.antialiasing();
+            scene.shadowFilter();
+            scene.physics();
+            scene.character();
+
+        }
+
+    }
+
+    /**
+     * Simple first person physical character control.
+     */
+    public class FirstPersonCharacter extends BetterCharacterControl implements AnalogListener {
```

Comment:
Decouple the `FirstPersonCharacter` and `BetterCharacterControl` logic. The current composition leads to an undesirable state where `BetterCharacterControl.setEnabled(false)` disables physics while `AnalogListener` continues to process input. Furthermore, for basic movement and jumping, an `ActionListener` might offer a more intuitive approach than the current `AnalogListener`.